### PR TITLE
[feat]: [DEL-4017]: API to generate helm-values.yaml file in NG

### PR DESCRIPTION
**Pull request by AnupamIO**
JIRA: https://harness.atlassian.net/browse/DEL-4017

What is Changed: Adding API that will generate helm values yaml file in NG.
Testing: Tested on localhost.

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)

### CODE REVIEW

**Review by arpit3107 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 6 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/client/DelegateNgManagerCgManagerClient.java` line 154 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rMGhi-RUIJK-lJ_HS:-N1rMGhi-RUIJK-lJ_HT:bml0x91) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/120-ng-manager/src/main/java/io/harness/ng/core/delegate/client/DelegateNgManagerCgManagerClient.java#L154)):*
> ```Java
>       @Query(DelegateMtlsApiConstants.API_PARAM_DOMAIN_PREFIX_NAME) @NotNull String domainPrefix);
> 
>   //------------------------ Delegate Setup New Apis -----------------------------------
> ```

NG Delegate Setup Apis



_Code quote:_
```Java
  //------------------------ Delegate Setup New Apis -----------------------------------
```
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 55 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rMYUkE4UuKHTT7Z2T:-N1rMYUl8G7dxGSWkXPo:bpvzg22) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L55)):*
> ```Java
> @Slf4j
> @OwnedBy(HarnessTeam.DEL)
> @Tag(name = "Delegate Setup Resource", description = "Contains APIs related to Delegate Setup in ng")
> ```

Remove ng from all these comments. these comments are used to generate ng docs. So, we should not use ng term again.



_Code quote:_
```Java
Contains APIs related to Delegate Setup in ng"
```
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 108 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rN5HlCI-46dHveyvs:-N1rN5HlCI-46dHveyvt:b2w1o2e) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L108)):*
> ```Java
>         accountIdentifier, orgIdentifier, projectIdentifier, delegateSetupDetails));
>     return Response.ok(delegateFile)
>         .header(CONTENT_TRANSFER_ENCODING, BINARY)
> ```

this return object is inconsistent with few apis that I am writing. Please wait for my pr to be merged and then rebase and update it. I will inform you.



_Code quote:_
```Java
    return Response.ok(delegateFile)
        .header(CONTENT_TRANSFER_ENCODING, BINARY)
        .type("text/plain; charset=UTF-8")
        .header(CONTENT_DISPOSITION, ATTACHMENT_FILENAME + HARNESS_DELEGATE_VALUES_YAML + YAML)
        .build();
```
___
*[`400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java` line 41 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rNah30AjC7imSlVUU:-N1rNah30AjC7imSlVUV:b-gxilh1) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java#L41)):*
> ```Java
> import lombok.extern.slf4j.Slf4j;
> 
> @Api("/delegate-setup/ng")
> ```

better name it as /delegate-setup/internal



_Code quote:_
```Java
@Api("/delegate-setup/ng")
@Path("/delegate-setup/ng")
```
___
*[`400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java` line 64 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rNgYk-mIwLWpoqwao:-N1rNgYk-mIwLWpoqwap:b-7pz9rc) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java#L64)):*
> ```Java
>   @ExceptionMetered
>   @InternalApi
>   public RestResponse<File> generateNgHelmValuesYaml(@Context HttpServletRequest request,
> ```

Have you tested it? I dont think you will be getting proper HttpServletRequest here and extract managerUrl and other info.



_Code quote:_
```Java
@Context HttpServletRequest request,
```
___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 1317 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rOOAMANVCcD7X6bN9:-N1rOOAMANVCcD7X6bNA:b-h8fg6r) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L1317)):*
> ```Java
>                       templateParameters.getAccountId(), templateParameters.getDelegateType()))
>               .put("accountId", templateParameters.getAccountId())
>               .put("nextGen", String.valueOf(isNgDelegate))
> ```

why do you need this? We have been generating ng yamls without this.



_Code quote:_
```Java
put("nextGen", String.valueOf(isNgDelegate))
```
___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 3997 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rOp-73x7iYxh29q5f:-N1rOp-73x7iYxh29q5g:b-jigde) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L3997)):*
> ```Java
>         true);
> 
>     File yaml = File.createTempFile(HARNESS_NG_DELEGATE, YAML);
> ```

make it consistent with other ng delegate methods. I think you need to define a new .yaml.ftl file for ng, and a corresponding test file also. And that should have INIT_SCRIPT also. Please file a ticket for INIT_SCRIPT in ng.



_Code quote:_
```Java
    File yaml = File.createTempFile(HARNESS_NG_DELEGATE, YAML);
    saveProcessedTemplate(scriptParams, yaml, "delegate-helm-values.yaml.ftl");
```

**Review by arpit3107 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 6 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 3952 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rQ6EU8FU9M86mQfmt:-N1rQ6EU8FU9M86mQfmu:bh314jj) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L3952)):*
> ```Java
>   public File generateNgHelmValuesYaml(String accountId, DelegateSetupDetails delegateSetupDetails, String managerHost,
>       String verificationServiceUrl) throws IOException {
>     validateKubernetesSetupDetails(accountId, delegateSetupDetails);
> ```

this should have a separate validateSetupDetails method for helm. Also, make sure you are checking uniqueness of delegateName.



_Code quote:_
```Java
validateKubernetesSetupDetails(accountId, delegateSetupDetails);
```

**Review by gaurav-jain1 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 6 file(s) to review. 2 change requests. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 73 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1s3YVKCvGbi4b92rT5:-N1s3YVKCvGbi4b92rT6:bubg2v1) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L73)):*
> ```Java
>   private static final String CONTENT_DISPOSITION = "Content-Disposition";
>   private static final String ATTACHMENT_FILENAME = "attachment; filename=";
>   public static final String YAML = ".yaml";
> ```

private ?



_Code quote:_
```Java
public
```
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 87 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1s4Eli5aeCiYY2yoVl:-N1s4Elj31YJAiAn1dDt:bbmqw9l) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L87)):*
> ```Java
>   @POST
>   @Timed
>   @Path("generate-helm-values-yaml")
> ```

We need to make this api to work with api-key as well, check with @arpit3107 
___
*[`400-rest/src/main/resources/delegatetemplates/delegate-helm-values.yaml.ftl` line 35 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1s5Mar9hMVp03GNV7T:-N1s5Mar9hMVp03GNV7U:bojn2cw) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/resources/delegatetemplates/delegate-helm-values.yaml.ftl#L35)):*
> ```FreeMarker
> delegateReplicas: ${delegateReplicas}
> delegateCpu: ${delegateCpu}
> delegateRam: ${delegateRam}
> ```

Do we need cpu and ram here ? how cg helm works without these values ?



_Code quote:_
```FreeMarker
delegateCpu: ${delegateCpu}
delegateRam: ${delegateRam}
```

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: Some of the checks are too old. 6 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/client/DelegateNgManagerCgManagerClient.java` line 154 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rMGhi-RUIJK-lJ_HS:-N2BjfH0AbFzYj9HV5px:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/120-ng-manager/src/main/java/io/harness/ng/core/delegate/client/DelegateNgManagerCgManagerClient.java#L154)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

NG Delegate Setup Apis
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 55 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rMYUkE4UuKHTT7Z2T:-N2BlKF04iK4Dzxql2X4:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L55)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

Remove ng from all these comments. these comments are used to generate ng docs. So, we should not use ng term again.
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 73 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1s3YVKCvGbi4b92rT5:-N2BlNduCCFasnZTXBTU:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L73)):*
<details><summary><i>Previously, gaurav-jain1 (Gaurav Jain) wrote…</i></summary><blockquote>

private ?
</blockquote></details>

Done.
___
*[`400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java` line 64 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rNgYk-mIwLWpoqwao:-N2BlTuK2rFhNV5LNh4D:b-62c6dg) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java#L64)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

Have you tested it? I dont think you will be getting proper HttpServletRequest here and extract managerUrl and other info.
</blockquote></details>

yes, tested on local setup.

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: Some of the checks are too old. 6 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 87 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1s4Eli5aeCiYY2yoVl:-N2Bzxkm9rof2V8HmjGI:b-i7cseb) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L87)):*
<details><summary><i>Previously, gaurav-jain1 (Gaurav Jain) wrote…</i></summary><blockquote>

We need to make this api to work with api-key as well, check with @arpit3107 
</blockquote></details>

checked with @arpit3107 this will work with apiKeys, no further changes required.

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: Some of the checks are too old. 6 file(s) to review. 2 change requests. 6. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java` line 41 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rNah30AjC7imSlVUU:-N2C-Tv66TOou69CgtqN:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java#L41)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

better name it as /delegate-setup/internal
</blockquote></details>

Done.

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: Some of the checks are too old. 6 file(s) to review. 2 change requests. 8. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 3997 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rOp-73x7iYxh29q5f:-N2DCXGW1wF57zD6TE4M:brbnsux) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L3997)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

make it consistent with other ng delegate methods. I think you need to define a new .yaml.ftl file for ng, and a corresponding test file also. And that should have INIT_SCRIPT also. Please file a ticket for INIT_SCRIPT in ng.
</blockquote></details>

done, as discussed I'll cover init_script along with helm chart changes and as part of: https://harness.atlassian.net/browse/DEL-4092

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: Some of the checks are too old. 6 file(s) to review. 2 change requests. 9. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/resources/delegatetemplates/delegate-helm-values.yaml.ftl` line 35 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1s5Mar9hMVp03GNV7T:-N2DD2_WDTa2UF2ND2tu:b-iz4sx3) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/resources/delegatetemplates/delegate-helm-values.yaml.ftl#L35)):*
<details><summary><i>Previously, gaurav-jain1 (Gaurav Jain) wrote…</i></summary><blockquote>

Do we need cpu and ram here ? how cg helm works without these values ?
</blockquote></details>

yes we need it, in cg values are fixed and hardcoded here: https://github.com/wings-software/helm-charts/blob/master/harness-delegate/templates/statefulset.yaml#L77-L78

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 11 file(s) to review. 2 change requests. 10. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 3952 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rQ6EU8FU9M86mQfmt:-N2DkRDm3d05hVI0XUNN:basjge) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L3952)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

this should have a separate validateSetupDetails method for helm. Also, make sure you are checking uniqueness of delegateName.
</blockquote></details>

validatingKubernetesSetup should also work for helm delegates as well because initial setup will be same for both in ng. helm will be just another way to install k8 delegate. Added a small check for helm_delegates in same method, please check.

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 11 file(s) to review. 2 change requests. 10. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 3997 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rOp-73x7iYxh29q5f:-N2DktYLDKNFcV-7On6V:b-l7o9bq) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L3997)):*
<details><summary><i>Previously, AnupamIO (Anupam Jaiswal) wrote…</i></summary><blockquote>

done, as discussed I'll cover init_script along with helm chart changes and as part of: https://harness.atlassian.net/browse/DEL-4092
</blockquote></details>

added mediaType parameter as well. However, as I understand mediaType other than text/plain will never be passed for ng delegate Apis. let me know your thoughts.

**Review by gaurav-jain1 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 11 file(s) to review. 2 change requests. 10. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/client/DelegateNgManagerCgManagerClient.java` line 161 at r2](https://reviewable.io/reviews/harness/harness-core/33091#-N2Pl43P-BwiyR_tG-FV:-N2Pl43Q1QwldRA3LEqQ:bu5zlvw) ([raw file](https://github.com/harness/harness-core/blob/9b7885764a19dd59f2a718e664dac53cc7a68b8f/120-ng-manager/src/main/java/io/harness/ng/core/delegate/client/DelegateNgManagerCgManagerClient.java#L161)):*
> ```Java
>       @Query(NGCommonEntityConstants.ACCOUNT_KEY) @NotNull String accountIdentifier,
>       @Query(NGCommonEntityConstants.ORG_KEY) String orgIdentifier,
>       @Query(NGCommonEntityConstants.PROJECT_KEY) String projectIdentifier, @Query("fileFormat") MediaType fileFormat,
> ```

can we get rid of this ?



_Code quote:_
```Java
@Query("fileFormat") MediaType fileFormat
```

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 11 file(s) to review. 2 change requests. 11. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/client/DelegateNgManagerCgManagerClient.java` line 161 at r2](https://reviewable.io/reviews/harness/harness-core/33091#-N2Pl43P-BwiyR_tG-FV:-N2XtoimBQR3qR7tp2YU:bnsgg8y) ([raw file](https://github.com/harness/harness-core/blob/9b7885764a19dd59f2a718e664dac53cc7a68b8f/120-ng-manager/src/main/java/io/harness/ng/core/delegate/client/DelegateNgManagerCgManagerClient.java#L161)):*
<details><summary><i>Previously, gaurav-jain1 (Gaurav Jain) wrote…</i></summary><blockquote>

can we get rid of this ?
</blockquote></details>

yes, this is not needed in ng, removed.

**Comment by AnupamIO** [comment]
trigger compile

**Review by gaurav-jain1 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 11 file(s) to review. 2 change requests. 11. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 53 at r6](https://reviewable.io/reviews/harness/harness-core/33091#-N3Pqokw8ZQ8x-1dzgIB:-N3Pqokx4r2T8iEugBYs:b-lvqo31) ([raw file](https://github.com/harness/harness-core/blob/8a0731ba39f9a52e141e8da19999578f59b31f78/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L53)):*
> ```Java
> 
> @Api("delegate-setup-ng")
> @Path("/delegate-setup-ng")
> ```

I would prefer using 'ng' as path component i.e. api/ng/delegate-setup/download-helm-values



_Code quote:_
```Java
elegate-setup-ng
```
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 93 at r6](https://reviewable.io/reviews/harness/harness-core/33091#-N3Pr41z5YR5PiPbZyDF:-N3Pr41z5YR5PiPbZyDG:b-vyzonr) ([raw file](https://github.com/harness/harness-core/blob/8a0731ba39f9a52e141e8da19999578f59b31f78/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L93)):*
> ```Java
>   @POST
>   @Timed
>   @Path("generate-helm-values-yaml")
> ```

consider renaming to `generate-helm-values`



_Code quote:_
```Java
generate-helm-values-yam
```

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 11 file(s) to review. 2 change requests. 13. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 53 at r6](https://reviewable.io/reviews/harness/harness-core/33091#-N3Pqokw8ZQ8x-1dzgIB:-N3TMBQf4Xt5lU_350YJ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8a0731ba39f9a52e141e8da19999578f59b31f78/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L53)):*
<details><summary><i>Previously, gaurav-jain1 (Gaurav Jain) wrote…</i></summary><blockquote>

I would prefer using 'ng' as path component i.e. api/ng/delegate-setup/download-helm-values
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 93 at r6](https://reviewable.io/reviews/harness/harness-core/33091#-N3Pr41z5YR5PiPbZyDF:-N3TMAUi4ROgCEqI7H-F:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8a0731ba39f9a52e141e8da19999578f59b31f78/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L93)):*
<details><summary><i>Previously, gaurav-jain1 (Gaurav Jain) wrote…</i></summary><blockquote>

consider renaming to `generate-helm-values`
</blockquote></details>

Done.

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: Some of the checks are too old. 11 file(s) to review. 2 change requests. 14. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 53 at r6](https://reviewable.io/reviews/harness/harness-core/33091#-N3Pqokw8ZQ8x-1dzgIB:-N3sYs0L4zAoI1oo0rU_:buwettg) ([raw file](https://github.com/harness/harness-core/blob/8a0731ba39f9a52e141e8da19999578f59b31f78/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L53)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

dont use api/ng because final url will be ng/api/ng/delegate-setup  which is not good. Use a different name
</blockquote></details>

changed as delegate-setup only.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 107 at r7](https://reviewable.io/reviews/harness/harness-core/33091#-N3Ujr0RDUUtU34Hm7cB:-N3sB0HD7hmzKbrGEZW1:b2x3phu) ([raw file](https://github.com/harness/harness-core/blob/149e90be93711522d045c304614113d7ece03ce6/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L107)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

We need a simpler body. This delegateSetupDetails has lots of unnecessary fields which are serialised to a yaml. Customers will not be sending those variables. Also, make sure to use default variables is those are not passed in body. For example delegateTokenName, delegateSize etc..
</blockquote></details>

The minimum fields that will be need to sent is:
{
    "name": "abc",
    "size": "LAPTOP",
    "k8sConfigDetails": {
        "k8sPermissionType": "NAMESPACE_ADMIN",
        "namespace": "harness-anupam"
    },
    "delegateType": "HELM_DELEGATE",
    "tokenName": "default_token"
}

Optional fields includes: tags, description, hostname, configurationId, org identifer, project identifier and identifier (groupname)

As this API will be used from UI and most of the above fields can be non-null hence I think it should be okay to keep request body same as delegateSetupDetails. Anyways we are creating new set of APIs for downloading delegates, I can create similar API for helm as well after you merge your code.

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 21 file(s) to review. 2 change requests. 15. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 108 at r1](https://reviewable.io/reviews/harness/harness-core/33091#-N1rN5HlCI-46dHveyvs:-N3szxgK0Wut6FQxtoZZ:b32wtd) ([raw file](https://github.com/harness/harness-core/blob/eed8ed3df96996f9402ed1971f51ca4910734452/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L108)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

this return object is inconsistent with few apis that I am writing. Please wait for my pr to be merged and then rebase and update it. I will inform you.
</blockquote></details>

ok

**Review by arpit3107 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 21 file(s) to review. 2 change requests. 15. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 107 at r7](https://reviewable.io/reviews/harness/harness-core/33091#-N3Ujr0RDUUtU34Hm7cB:-N3u5JLHBEBRwEYoKkNw:b-tki6sg) ([raw file](https://github.com/harness/harness-core/blob/149e90be93711522d045c304614113d7ece03ce6/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L107)):*
<details><summary><i>Previously, AnupamIO (Anupam Jaiswal) wrote…</i></summary><blockquote>

The minimum fields that will be need to sent is:
{
    "name": "abc",
    "size": "LAPTOP",
    "k8sConfigDetails": {
        "k8sPermissionType": "NAMESPACE_ADMIN",
        "namespace": "harness-anupam"
    },
    "delegateType": "HELM_DELEGATE",
    "tokenName": "default_token"
}

Optional fields includes: tags, description, hostname, configurationId, org identifer, project identifier and identifier (groupname)

As this API will be used from UI and most of the above fields can be non-null hence I think it should be okay to keep request body same as delegateSetupDetails. Anyways we are creating new set of APIs for downloading delegates, I can create similar API for helm as well after you merge your code.
</blockquote></details>

Yes, I totally agree with you. But @mdmilic made me to create a new request body and use it. Hence you have to do it 🤣.
One more thing, customers must not send delegateType in request body. You have to handle that internally.

**Review by mdmilic — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 21 file(s) to review. 2 change requests. 15. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 107 at r7](https://reviewable.io/reviews/harness/harness-core/33091#-N3Ujr0RDUUtU34Hm7cB:-N3xoP5_1zECPndhIYLp:b-w77iq8) ([raw file](https://github.com/harness/harness-core/blob/149e90be93711522d045c304614113d7ece03ce6/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L107)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

Yes, I totally agree with you. But @mdmilic made me to create a new request body and use it. Hence you have to do it 🤣.
One more thing, customers must not send delegateType in request body. You have to handle that internally.
</blockquote></details>

Even though it's not ideal, as long as this is UI only API it's ok to use delegateSetupDetails. For customers we need to create something simple and more specific. So I would say keep it here and create a separate API that customers could use. We could eventually even transition the UI to the APIs customers use (as long as that doesn't mean we bloat it again).

**Comment by AnupamIO** [comment]
trigger ti1

**Comment by AnupamIO** [comment]
trigger ti4

**Comment by AnupamIO** [comment]
trigger ti0

**Review by gaurav-jain1 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: 21 file(s) to review. 1 change requests. 11. unresolved discussions.

**Review by arpit3107 — APPROVED** [review]
> Reviewed 1 of 8 files at r3, 1 of 4 files at r5, 7 of 12 files at r8, 5 of 6 files at r9, 6 of 7 files at r10, 1 of 1 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: Some of the checks are too old. 2. unresolved discussions.

**Comment by AnupamIO** [comment]
trigger recency

**Review by gaurav-jain1 — APPROVED** [review]
> Reviewed 1 of 4 files at r5, 1 of 12 files at r8, 4 of 6 files at r9, 2 of 7 files at r10, 1 of 1 files at r11, 6 of 6 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33091)* status: :shipit: complete! ✓
